### ISSUE TITLE
storcli controller 0 not fount

### ISSUE THREAD

**Issue by javagyt** [thomas-krenn/check_lsi_raid#26]
I use Meagecli64 command:
`/opt/MegaRAID/MegaCli/MegaCli64 -AdpAllInfo -a0`
I can get RAID controller information,for example:
```
Adapter #0

==============================================================================
                    Versions
                ================
Product Name    : PERC H730P Mini
Serial No       : 72O00BI
FW Package Build: 25.5.0.0018

                    Mfg. Data
                ================
Mfg. Date       : 02/27/17
Rework Date     : 02/27/17
Revision No     : A06
Battery FRU     : N/A

                Image Versions in Flash:
                ================
BIOS Version       : 6.33.01.0_4.16.07.00_0x06120200
Ctrl-R Version     : 5.18-0700
FW Version         : 4.270.00-8112
NVDATA Version     : 3.1511.00-0014
Boot Block Version : 3.07.00.00-0003

                Pending Images in Flash
                ================
None

                PCI Info
                ================
Controller Id	: 0000
Vendor Id       : 1000
Device Id       : 005d
SubVendorId     : 1028
SubDeviceId     : 1f47

Host Interface  : PCIE
```
but now,I want to use storcli:
```
[root@node-4 ~]# /opt/MegaRAID/storcli/storcli64 /c0 show all
CLI Version = 007.0913.0000.0000 Jan 11, 2019
Operating system = Linux 3.10.0-693.11.1.el7.es.11.x86_64
Controller = 0
Status = Failure
Description = Controller 0 not found
```
my raid controller is PERC H730P Mini from DELL

**Comment by tk-wfischer** [comment]
I'm not familiar with the DELL controllers and their firmware changes on the Broadcom/LSI chips. The plugin only works with storcli.

- Can you find any information regarding storcli support on the Dell support pages for your controller?
- Does Dell maybe provide an altered version of storcli which might work with your controller?

**Comment by javagyt** [comment]
I can not find any information regarding storcli,I discovered DELL have their own tool  perccli to manage controller.

**Comment by tk-wfischer** [comment]
Could it be that perccli is just their name for the tool which has the same parameters like storcli? Could you try a simple symbolic link so that /usr/sbin/storcli is a symlink to the perccli tool?

**Comment by javagyt** [comment]
I install perccli just now:
[root@node-4 Linux]# /opt/MegaRAID/perccli/perccli64 show
```
Status Code = 0
Status = Success
Description = None

Number of Controllers = 2
Host Name = node-4
Operating System  = Linux3.10.0-693.11.1.el7.es.11.x86_64

System Overview :
===============

---------------------------------------------------------------------------
Ctl Model           Ports PDs DGs DNOpt VDs VNOpt BBU sPR DS EHS ASOs Hlth 
---------------------------------------------------------------------------
  0 PERCH730PMini       8  22  17     0  17     0 Opt On  3  N      0 Opt  
  1 PERCH830Adapter     8   0   0     0   0     0 Opt On  3  N      0 Opt  
---------------------------------------------------------------------------

Ctl=Controller Index|DGs=Drive groups|VDs=Virtual drives|Fld=Failed
PDs=Physical drives|DNOpt=DG NotOptimal|VNOpt=VD NotOptimal|Opt=Optimal
Msng=Missing|Dgd=Degraded|NdAtn=Need Attention|Unkwn=Unknown
sPR=Scheduled Patrol Read|DS=DimmerSwitch|EHS=Emergency Hot Spare
Y=Yes|N=No|ASOs=Advanced Software Options|BBU=Battery backup unit
Hlth=Health|Safe=Safe-mode boot

```
this tool command the same as storcli，I even suspect that perccli copied storcli,it just rename.

**Comment by tk-wfischer** [comment]
Great - thank you for the feedback.
- Does the plugin work now for you?
- In case everything is fine I would close this issue.

**Comment by javagyt** [comment]
yes，thank you!

**Comment by javagyt** [comment]
hi，               
                                 my name is javagyt，we discussed some problems about raid in github.               
                                 I recently had a problem.my disk automatic change  Foreign State: Foreign in raid Controller.I use command "MegaCli64  -cfgforeign -Import -a0",this disk can be online,but,it will be foreign a  few minutes later.      
                                            
               
                            Enclosure Device ID: 32
Slot Number: 4
Enclosure position: N/A
Device Id: 4
WWN: 
Sequence Number: 7
Media Error Count: 0
Other Error Count: 1
Predictive Failure Count: 0
Last Predictive Failure Event Seq Number: 0
PD Type: SATA

Raw Size: 931.512 GB [0x74706db0 Sectors]
Non Coerced Size: 931.012 GB [0x74606db0 Sectors]
Coerced Size: 931.0 GB [0x74600000 Sectors]
Firmware state: Unconfigured(good), Spun Up
Device Firmware Level: SDM1
Shield Counter: 0
Successful diagnostics completion on :  N/A
SAS Address(0): 0x1221000004000000
Connected Port Number: 4(path0) 
Inquiry Data:             WL1DMNC9ST1000LM048-2E7172                      SDM1    
FDE Capable: Not Capable
FDE Enable: Disable
Secured: Unsecured
Locked: Unlocked
Needs EKM Attention: No
Foreign State: Foreign 
Foreign Secure: Drive is not secured by a foreign lock key
Device Speed: Unknown 
Link Speed: Unknown 
Media Type: Hard Disk Device
Drive Temperature : N/A
PI Eligibility:  No 
Drive is formatted for PI information:  No
PI: No PI
Port-0 :
Port status: Active
Port's Linkspeed: Unknown 
Drive has flagged a S.M.A.R.T alert : No


I use smartctl to check:
[root@node-1 ~]# smartctl -d sat+megaraid,4 -H /dev/sde
smartctl 6.2 2017-02-27 r4394 [x86_64-linux-3.10.0-514.16.1.el7.4.x86_64] (local build)
Copyright (C) 2002-13, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF READ SMART DATA SECTION ===
SMART overall-health self-assessment test result: PASSED
Warning: This result is based on an Attribute check.


[root@node-1 ~]# smartctl -d sat+megaraid,4 -A /dev/sde
smartctl 6.2 2017-02-27 r4394 [x86_64-linux-3.10.0-514.16.1.el7.4.x86_64] (local build)
Copyright (C) 2002-13, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF READ SMART DATA SECTION ===
SMART Attributes Data Structure revision number: 10
Vendor Specific SMART Attributes with Thresholds:
ID# ATTRIBUTE_NAME          FLAG     VALUE WORST THRESH TYPE      UPDATED  WHEN_FAILED RAW_VALUE
  1 Raw_Read_Error_Rate     0x000f   055   055   006    Pre-fail  Always       -       83400025
  3 Spin_Up_Time            0x0003   099   099   000    Pre-fail  Always       -       0
  4 Start_Stop_Count        0x0032   100   100   020    Old_age   Always       -       11
  5 Reallocated_Sector_Ct   0x0033   095   095   036    Pre-fail  Always       -       3088
  7 Seek_Error_Rate         0x000f   082   060   045    Pre-fail  Always       -       142063637
  9 Power_On_Hours          0x0032   095   095   000    Old_age   Always       -       4416 (178 191 0)
 10 Spin_Retry_Count        0x0013   100   100   097    Pre-fail  Always       -       0
 12 Power_Cycle_Count       0x0032   100   100   020    Old_age   Always       -       11
184 End-to-End_Error        0x0032   100   100   099    Old_age   Always       -       0
187 Reported_Uncorrect      0x0032   060   060   000    Old_age   Always       -       40
188 Command_Timeout         0x0032   099   099   000    Old_age   Always       -       17180131332
189 High_Fly_Writes         0x003a   100   100   000    Old_age   Always       -       0
190 Airflow_Temperature_Cel 0x0022   073   068   040    Old_age   Always       -       27 (Min/Max 25/32)
191 G-Sense_Error_Rate      0x0032   100   100   000    Old_age   Always       -       0
192 Power-Off_Retract_Count 0x0032   100   100   000    Old_age   Always       -       3
193 Load_Cycle_Count        0x0032   050   050   000    Old_age   Always       -       100553
194 Temperature_Celsius     0x0022   027   040   000    Old_age   Always       -       27 (0 11 0 0 0)
197 Current_Pending_Sector  0x0012   099   099   000    Old_age   Always       -       152
198 Offline_Uncorrectable   0x0010   099   099   000    Old_age   Offline      -       152
199 UDMA_CRC_Error_Count    0x003e   200   200   000    Old_age   Always       -       0
240 Head_Flying_Hours       0x0000   100   253   000    Old_age   Offline      -       93681826664447
241 Total_LBAs_Written      0x0000   100   253   000    Old_age   Offline      -       29359425170
242 Total_LBAs_Read         0x0000   100   253   000    Old_age   Offline      -       18883843428
254 Free_Fall_Sensor        0x0032   100   100   000    Old_age   Always       -       0






               
                              




------------------ 原始邮件 ------------------
发件人: "Werner Fischer"<notifications@github.com>;
发送时间: 2019年4月23日(星期二) 下午5:55
收件人: "thomas-krenn/check_lsi_raid"<check_lsi_raid@noreply.github.com>;
抄送: "锐特软科技"<245457993@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [thomas-krenn/check_lsi_raid] storcli controller 0 not fount(#26)




Closed #26.
